brackend/endpoints/tournaments/repositories/TournamentRepository.py

```diff
@@ -0,0 +1,22 @@
+from abc import ABC
+from sqlalchemy.orm import Session
+from brackend.db.models import (
+    EngineGetter,
+    Tournament,
+)
+
+
+class TournamentRepository(ABC):
+    engine = EngineGetter.get_or_create_engine()
+
+    @classmethod
+    def search_by_name(cls, name, count=20):
+        with Session(cls.engine) as session:
+            search = "%{}%".format(name)
+            results = session.query(Tournament)\
+                .filter(Tournament.name.like(search))\
+                .limit(count)\
+                .all()
+            return results
+
+
```

Comment:
This is the pattern I'd recommend we move towards when doing synchronous actions, rather than using tasks for all DB queries